clients/python/provider/superposition_provider/local_provider.py

```
@@ -487,6 +505,68 @@ async def _start_refresh_strategy(self) -> None:
+                                                logger.info("Provider garbage collected, stopping SSE loop.")
+                                                return
+                                            line = line_bytes.decode("utf-8", errors="replace").strip()
+                                            logger.debug(f"SSE raw line: {line!r}")
+                                            if not line or line.startswith(":"):
+                                                continue
+                                            if line.startswith("event:") or line.startswith("data:"):
+                                                async def _do_refresh(ref=self_ref):
+                                                    await asyncio.sleep(debounce_s)
+                                                    try:
+                                                        await ref.refresh()
+                                                    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)